include/swift/Basic/CXXStdlibKind.h

```diff
@@ -0,0 +1,30 @@
+//===--- CXXStdlibKind.h ----------------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_BASIC_CXX_STDLIB_KIND_H
+#define SWIFT_BASIC_CXX_STDLIB_KIND_H
+
+namespace swift {
+
+enum class CXXStdlibKind : uint8_t {
+  /// Use the default C++ stdlib for a particular platform: libc++ for Darwin,
+  /// libstdc++ for most Linux distros.
+  PlatformDefault = 0,
```

Comment:
I think that we shouldn't just use a blanket "PlatformDefault" .. what if I want to use libstdc++ on a libc++ defaulting platform? We really should enumerate the C++ runtimes IMO:

1. msvcprt
2. libstdc++
3. libc++
4. stl_port
5. stdc++
6. dinkumware
7. eastl

are ones off the top of my head that I can think of as being quite common.

Reply:
I'd like to keep this patch focused on a single use case, libc++ on Linuxes that use libstdc++ by default. I'm trying to keep the implementation fairly generic, to make support of other stdlibs a potential future improvement. But I'm not familiar with the rest of the libraries you've listed and I don't really have an easy way to test them.

> what if I want to use libstdc++ on a libc++ defaulting platform?

In that case we would need to add `OverrideLibstdcxx` to this enum.

Reply:
Having them be enumerated and not used should be fine though. I don't think that you need to implement support for all of them. I'm just trying to show that "platform default" doesn't convey enough information.

Reply:
How about renaming `PlatformDefault` to `ClangDefault`?

Reply:
I changed this to list the supported C++ stdlibs explicitly: libc++, libstdc++, msvcprt. I also removed the platform-default option to make this logic easier to understand.

I agree that we shouldn't list the stdlibs that we don't actually support.